freqtrade/optimize/backtesting.py

```diff
@@ -253,22 +253,19 @@ def _get_sell_trade_entry(
             sell = self.strategy.should_sell(trade, sell_row.open, sell_row.date, sell_row.buy,
                                              sell_row.sell, low=sell_row.low, high=sell_row.high)
             if sell.sell_flag:
-
                 trade_dur = int((sell_row.date - buy_row.date).total_seconds() // 60)
                 # Special handling if high or low hit STOP_LOSS or ROI
                 if sell.sell_type in (SellType.STOP_LOSS, SellType.TRAILING_STOP_LOSS):
                     # Set close_rate to stoploss
                     closerate = trade.stop_loss
                 elif sell.sell_type == (SellType.ROI):
-                    # get next entry in min_roi > to trade duration
-                    # Interface.py skips on trade_duration <= duration
-                    roi_entry = max(list(filter(lambda x: trade_dur >= x,
-                                                self.strategy.minimal_roi.keys())))
-                    roi = self.strategy.minimal_roi[roi_entry]
-
-                    # - (Expected abs profit + open_rate + open_fee) / (fee_close -1)
-                    closerate = - (trade.open_rate * roi + trade.open_rate *
-                                   (1 + trade.fee_open)) / (trade.fee_close - 1)
+                    roi = self.strategy.min_roi_reached_entry(trade_dur)
+                    if roi is not None:
+                        # - (Expected abs profit + open_rate + open_fee) / (fee_close -1)
+                        closerate = - (trade.open_rate * roi + trade.open_rate *
+                                       (1 + trade.fee_open)) / (trade.fee_close - 1)
+                    else:
+                        closerate = sell_row.open
```

Comment:
I don't think this line (or the else branch in total) can be hit at all.
Remember that it's within the `if sell.sell_type == (SellType.ROI):` block.

Hitting this line would assume that `should_sell()` reports sell-reason as ROI - but `min_roi_reached_entry()` does not have a ROI for this "open" time - so this would indicate a huge bug somewhere in should_sell (which i don't think is there atm), since `should_sell()` uses `min_roi_reached_entry()` too.

I just tested this by raising an exception on this line (and trying different ROI configurations, including an empty dict) - and this was never hit, so I'm not sure if we shouldn't remove this line completely - since it's unreachable code.

Reply:
> I don't think this line (or the else branch in total) can be hit at all.

It will be hit in exactly same conditions when the user got previously exception in #1948 
(previously it was an unexpected exception generated by max(), now it's None returned as min_roi entry val, it's handled above in the calling function, so some closerate should be assigned to proceed...)

Reply:
maybe i misunderstand #1948 - but wasn't that a ROI dict without a 0 entry:
e.g.

```python
minimal_roi = { "200": "0.01"}
```

Reply:
> maybe i misunderstand #1948 - but wasn't that a ROI dict without a 0 entry:
> e.g.
> 
> ```python
> minimal_roi = { "200": "0.01"}
> ```

Yes and that line assigning the `closerate` should be hit when `trade_dur < 200`, i.e. when `None` is returned from `self.strategy.min_roi_reached_entry(trade_dur)`. I assume.

Reply:
well that's true for the `interface.py` file - but not here, since this whole code-section is ONLY hit if the sell-reason is ROI.

So if tradedur < 200 - then there is 
a) no sell or
b)  sell-reason is something else (stoploss / sell-signal, ...)

however this part won't be reached since for that to happen, you'd have to have sell_reason=ROI + trade_dur < 200 - (but then sell-reason won't be ROI)...